### PR TITLE
[wasi] CI: Provision `wasi-sdk`, and `wasmtime` only when needed

### LINKED ISSUE #82391: [wasi] Respect `WASMTIME_PATH`
we need to use wasmruntime of the CI image when available (and perhaps same version)
The [build](https://dev.azure.com/dnceng-public/public/_build/results?buildId=178139&view=logs&j=f02b8cf9-dd4d-54fc-c292-2bb1d305b019&t=65a12146-2c6c-404a-b052-28d912f15705&l=93) here is downloading wasmtime `5.0.0` 
It's image is `mcr.microsoft.com/dotnet-buildtools/prereqs:ubuntu-20.04-webassembly` which should have `3.0.0` on it. 

The image sets `WASMTIME_PATH` and `WASMTIME_VERSION` but we don't use it.

@radical @radekdoulik

### PR DESCRIPTION
**Pull request by radical**
## Provision wasmtime, wasi-sdk on CI only if needed

On the linux VMs we have `wasi-sdk`, and `wasmtime` provisioned already. With this PR
the jobs will use those instead of downloading fresh copies for every build.

##     Add generic support for staging helix payloads

For a directory specified as a helix correlation payload, the helix
tasks will attempt to write a `.payload` file to the directory. But if the
directory is not writable then we need to stage the payload in a
writable location, before passing the path to helix.

For example, if we have `wasi-sdk` installed on the system as:
    `WASI_SDK_PATH=/usr/local/wasi-sdk`

.. then we can stage it by copying to
`artifacts/obj/helix-staging/wasi-sdk`, and then using that staging path
for the helix correlation payload.

This was already being done for `EMSDK`, and `WasiSdk` in two separate
targets. And now needs to be done for `wasmtime` also. Instead of having
completely separate targets for all this, add a new mechanism to specify
helix dependencies to stage.

For example:
```xml
  <ItemGroup>
    <HelixDependenciesToStage Condition="'$(NeedsWasmtime)' == 'true'" SourcePath="$(WasmtimeDir)"   Include="$(WasmtimeDirForHelixPayload)" />
  </ItemGroup>
```

.. and the payload can use specified as
```xml
<HelixCorrelationPayload Include="$(WasmtimeDirForHelixPayload)" Destination="wasmtime" Condition="'$(NeedsWasmtime)' == 'true'" />
```

Fixes https://github.com/dotnet/runtime/issues/82391

### CODE REVIEW

**Review by lewing — APPROVED** [review]

File: src/libraries/sendtohelix-wasm.targets [hunk 1]
```diff
@@ -70,6 +81,15 @@
     <_HelixLocalNodePath Condition="'$(NeedsEMSDK)' == 'true' and '$(WindowsShell)' == 'true'">%HELIX_CORRELATION_PAYLOAD%\build\emsdk\node</_HelixLocalNodePath>
   </PropertyGroup>
 
+  <ItemGroup>
+    <HelixDependenciesToStage
```
Comment (pavelsavara) [inline src/libraries/sendtohelix-wasm.targets:85]: why there is no WASI SDK here ? We don't need any ?
why there is no wasmtime here ?

Comment (pavelsavara) [reply src/libraries/sendtohelix-wasm.targets:85]: Scratch it, I see it above now.

**Comment by radical** [comment]
Failures are unrelated, and known.

**Review by steveisok — APPROVED** [review]

**Comment by radical** [comment]
Remaining builds are completely unrelated. Merging this early to free up some resources.

**Pull request merged**